base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
```

Comment:
I just saw that the `unique` methods are defined in `set.jl` and `multidimensional`; the reason seems to be that they are defined right after the needed tools (e.g. `Set` or multidim tools) are defined. This would make much more sense to me put those functions also in an "algorithm.jl" file. E.g. I would expect to find in "set.jl" only the functions implementing the interface of `Set`, and possibly methods specialized for `Set`, but functions merely using a set in their implementation have nothing to do there.